static/js/rename.js

```diff
@@ -0,0 +1,80 @@
+"use strict";
+window.onload = function() {
+  loadDeviceDetails();
+
+  if (hasGroupList()) {
+    var groups = getGroups();
+    var groupsSelect = document.getElementById("group-list")
+    for (var i in groups) {
+      var group = groups[i];
+      var groupOpt = document.createElement('option');
+      groupOpt.value = group;
+      groupOpt.innerHTML = group;
+      groupsSelect.appendChild(groupOpt);
+    }
+  }
+};
+
+function loadDeviceDetails() {
+  var xmlHttp = new XMLHttpRequest();
+  xmlHttp.open("GET", "/api/device-info", true);
+  xmlHttp.setRequestHeader("Authorization", "Basic "+btoa("admin:feathers"))
+
+  xmlHttp.onload = async function() {
+    if (xmlHttp.status == 200) {
+      var response = JSON.parse(xmlHttp.response);
+      document.getElementById("new-name").placeholder=response.devicename;
+      document.getElementById("new-group").placeholder=response.groupname;
+    } else {
+      console.log("error with getting device details");
+      console.log(xmlHttp);
+    }
+  }
+
+  xmlHttp.onerror = async function() {
+    console.log("error with getting device details");
+    console.log(xmlHttp);
+  }
+
+  xmlHttp.send(null)
+}
+
+function rename() {
+  var updateButton = document.getElementById("rename-button");
+  updateButton.disabled = true;
+  updateButton.innerHTML = "Renaming";
+
+  var data = new FormData();
+  data.append('name', document.getElementById("new-name").value);
+  data.append('group', document.getElementById("new-group").value);
```

Comment:
If you give the `<form>` and id then jQuery can turn the fields into the correctly formatted form data for you: `$( "#some-form" ).serialize()`